compiler/src/main/scala/org/scalajs/core/compiler/GenJSCode.scala

```diff
@@ -734,13 +733,36 @@ abstract class GenJSCode extends plugins.PluginComponent
         if (scalaPrimitives.isPrimitive(sym)) {
           None
         } else if (sym.isDeferred || sym.owner.isInterface) {
+          val body = if (sym.hasAnnotation(JavaDefaultMethodAnnotation)) {
+            /* For an interface method with @JavaDefaultMethod, make it a
+             * default method calling the impl class method.
+             */
```

Comment:
Exactly. Not to mention some other code could still call the impl class method body *statically*, so we can't just remove it altogether. They both need to exist, one being a forwarder to the other. We might as well keep the impl class version as the full version, since it already exists as is. At the end of the day, the optimizer removes all these indirections.

When scalac will emit default methods, we'll just have to make sure *not* to do this, but just emit the default methods. The specifics will depend on how scalac gives default methods to us, so it is fruitless to try and anticipate the implementation at the moment.